h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/ProxyStarter.java

```diff
@@ -21,7 +22,23 @@ public static String start(String[] args, JettyProxy.Credentials credentials, St
     JettyProxy proxy = initializeProxy(baseArgs, credentials, proxyTo);
 
     InetAddress address = HostnameGuesser.findInetAddressForSelf(baseArgs.ip, baseArgs.network);
-    return H2O.getURL(proxy.getScheme(), address, proxy.getPort(), baseArgs.context_path);
+    if (useHostname) {
+      String hostname = localIpToHostname(address);
+      return H2O.getURL(proxy.getScheme(), hostname, proxy.getPort(), baseArgs.context_path);
+    } else
+      return H2O.getURL(proxy.getScheme(), address, proxy.getPort(), baseArgs.context_path);
+  }
+
+  private static String localIpToHostname(InetAddress address) {
+    String hostname = address.getHostName();
+    if (! address.getHostAddress().equals(hostname))
+      return hostname;
+    // we don't want to return IP address (because of a security policy of a particular customer, see PUBDEV-5680)
+    hostname = System.getenv("HOSTNAME");
```

Comment:
Maybe include the reference to `HOSTNAME` env variable in the error message if the host-name could not be found? It would be good for debugging if needed

Reply:
Good suggestion! Thank you